lib/db.js

```diff
@@ -255,8 +256,9 @@ class LimitDBRedis extends EventEmitter {
   /**
    * Take N elements from a bucket if available.
    *
-   * @param {takeParams} params - The params for take.
-   * @param {function(Error, takeResult)} callback.
+   * @param {TakeParams} params
+   * @param {function(Error)} callback.
+   * @param {function(null, TakeResult)} callback.
```

Comment:
I prefer explicitly defining 2 signatures because:
1. using a union only for `Error` means that `callback(error)` is not allowed. You'd need `function(Error|null, TakeResult|null)` and then `callback()` is allowed.
2. The intended call patterns are explicit and obvious

Reply:
Ah good point, gotcha!